File: library/bottomsheets/src/main/java/com/nasahapps/mdt/bottomsheets/BottomSheetAdapter.java
package com.nasahapps.mdt.bottomsheets;

import android.content.DialogInterface;
import android.graphics.drawable.Drawable;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.nasahapps.materialdesigntoolbox.Utils;

import java.util.List;

/**
 * Created by hhasan on 10/14/16.
 */

abstract class BottomSheetAdapter extends RecyclerView.Adapter<BottomSheetAdapter.ViewHolder> {

    private List<BottomSheetItem> mItems;
    private DialogInterface mDialog;
    private OnClickListener mListener;
    private int mIconTint;

    private BottomSheetAdapter() {
    }

    BottomSheetAdapter(List<BottomSheetItem> items, DialogInterface dialog, OnClickListener listener) {
        mItems = items;
        mDialog = dialog;
        mListener = listener;
    }

    void setIconTint(int iconTint) {
        mIconTint = iconTint;
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        BottomSheetItem item = mItems.get(position);
        holder.text.setText(item.getText());
        Drawable icon = item.getIcon();
        if (icon != null && mIconTint != 0) {
            icon = Utils.getTintedDrawableCompat(icon, mIconTint);
        }
        holder.icon.setImageDrawable(icon);
    }

    @Override
    public int getItemCount() {
        return mItems.size();
    }

    public interface OnClickListener {
        void onClick(DialogInterface dialog, int position, int id);
    }

    class ViewHolder extends RecyclerView.ViewHolder {

        TextView text;
        ImageView icon;

        public ViewHolder(View v) {
            super(v);

            text = (TextView) v.findViewById(R.id.text);
            icon = (ImageView) v.findViewById(R.id.icon);

            v.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if (mListener != null) {
                        mListener.onClick(mDialog, getAdapterPosition(), mItems.get(getAdapterPosition()).getMenuId());
                    }
                }
            });
        }

    }

}

File: library/bottomsheets/src/main/java/com/nasahapps/mdt/bottomsheets/BottomSheetAdapter.kt
package com.nasahapps.mdt.bottomsheets

import android.content.DialogInterface
import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import com.nasahapps.mdt.Utils

/**
 * Created by hhasan on 10/14/16.
 */

abstract class BottomSheetAdapter constructor(private val mItems: List<BottomSheetItem>,
                                              private val mDialog: DialogInterface,
                                              private val mListener: OnClickListener?) : RecyclerView.Adapter<BottomSheetAdapter.ViewHolder>() {

    private var mIconTint: Int = 0

    fun setIconTint(iconTint: Int) {
        mIconTint = iconTint
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = mItems[position]
        holder.text.text = item.text
        var icon = item.icon
        if (icon != null && mIconTint != 0) {
            icon = Utils.getTintedDrawable(icon, mIconTint)
        }
        holder.icon.setImageDrawable(icon)

        holder.itemView.setOnClickListener {
            mListener?.onClick(mDialog, holder.adapterPosition, mItems[holder.adapterPosition].menuId)
        }
    }

    override fun getItemCount(): Int {
        return mItems.size
    }

    interface OnClickListener {
        fun onClick(dialog: DialogInterface, position: Int, id: Int)
    }

    class ViewHolder(v: View) : RecyclerView.ViewHolder(v) {

        var text = v.findViewById<TextView>(R.id.text)
        var icon = v.findViewById<ImageView>(R.id.icon)

    }

}
